### ISSUE TITLE
DropdownToggle 'aria-haspopup' PropType doesn't allow 'listbox'

### ISSUE THREAD

**Issue by illandril** [reactstrap/reactstrap#2208]
- components: `DropdownToggle`
- reactstrap version `#8.9.0`
- import method `esm`
- react version `#17.0.2`
- bootstrap version `#5.0.0`

### What is happening?

PropTypes validation for the `aria-haspopup` attribute for `DropdownToggle` fails if `listbox` is specified.

### What should be happening?

PropTypes validation shouldn't complain when `aria-haspopup` is set to `listbox`.

The `aria-haspopup` property should continue to accept the boolean `true`, which is appropriate for menu-like Dropdowns. I can't currently think of any reason why someone would want any other role for the Dropdown styling other than `listbox` or `menu` (`true` covers the `menu` case since that is an alias for `menu`), but I also don't see any harm in making the PropType definition allow any of the currently valid values for `aria-haspopup` attribute (or even any string at all).

### Steps to reproduce issue

1. Create a `Dropdown` that mimics the behavior of a `select`
2. Give the `DropdownToggle` an `aria-haspopup` of `listbox`
3. Give the `DropdownMenu` a `role` of `listbox`
4. Run in dev mode, so PropTypes validation errors are reported

### Error message in console

```
Warning: Failed prop type: Invalid prop `aria-haspopup` of type `string` supplied to `DropdownToggle`, expected `boolean`.
    at DropdownToggle (...)
    at Dropdown (...)
    at ...
```

### Code

```
export default function Example() {
  const options = ['Option A', 'Option B', 'Option C', 'Option D'];
  const [value, setValue] = useState(options[0]);
  return (
    <UncontrolledDropdown>
      <DropdownToggle caret aria-haspopup="listbox">
        {value}
      </DropdownToggle>
      <DropdownMenu role="listbox">
        {options.map((option, index) => (
          <DropdownItem key={index} onClick={() => setValue(option)}>
            {option}
          </DropdownItem>
        ))}
      </DropdownMenu>
    </UncontrolledDropdown>
  );
}
```

**Comment by JosephBrooksbank** [comment]
#2259 fixed this behavior by adding a `menuRole: boolean | string` prop to the `Dropdown` component. The `aria-haspopup` prop on `DropdownToggle` is no longer recommended, it seems
